contrib/tools/dummysigner/src/sign.rs

```diff
@@ -0,0 +1,204 @@
+use revault_tx::bitcoin::{
+    secp256k1,
+    util::{
+        bip143::SigHashCache,
+        bip32::{DerivationPath, ExtendedPrivKey},
+        psbt::PartiallySignedTransaction,
+    },
+    SigHashType,
+};
+use serde::{de, Deserialize, Deserializer, Serialize};
+use std::str::FromStr;
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(untagged)]
+pub enum SignRequest {
+    RevocationTransactions {
+        #[serde(deserialize_with = "deserialize_fromstr")]
+        derivation_path: DerivationPath,
+        target: RevocationTransactions,
+    },
+    UnvaultTransaction {
+        #[serde(deserialize_with = "deserialize_fromstr")]
+        derivation_path: DerivationPath,
+        target: UnvaultTransaction,
+    },
+    SpendTransaction {
+        /// vec of derivation path following the order of each spend psbt input
+        #[serde(deserialize_with = "deserialize_vec_fromstr")]
+        derivation_paths: Vec<DerivationPath>,
+        target: SpendTransaction,
+    },
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RevocationTransactions {
+    #[serde(with = "bitcoin_psbt")]
+    pub cancel_tx: PartiallySignedTransaction,
+
+    #[serde(with = "bitcoin_psbt")]
+    pub emergency_tx: PartiallySignedTransaction,
+
+    #[serde(with = "bitcoin_psbt")]
+    pub emergency_unvault_tx: PartiallySignedTransaction,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct UnvaultTransaction {
+    #[serde(with = "bitcoin_psbt")]
+    pub unvault_tx: PartiallySignedTransaction,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SpendTransaction {
+    #[serde(with = "bitcoin_psbt")]
+    pub spend_tx: PartiallySignedTransaction,
+}
+
+mod bitcoin_psbt {
+    use revault_tx::bitcoin::{consensus::encode, util::psbt::PartiallySignedTransaction};
+    use serde::{self, Deserialize, Deserializer, Serializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<PartiallySignedTransaction, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        let bytes: Vec<u8> = base64::decode(&s).map_err(serde::de::Error::custom)?;
+        encode::deserialize(&bytes).map_err(serde::de::Error::custom)
+    }
+
+    pub fn serialize<'se, S>(
+        psbt: &PartiallySignedTransaction,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&base64::encode(encode::serialize(&psbt)))
+    }
+}
```

Comment:
You know that we have code that does that in revault_tx, right? :sweat_smile:

Reply:
I am not sure where it is. I would still prefer to keep this code and not rely on an external dep.